lib/FusionInventory/Agent/Task/Inventory/Win32/License.pm

```diff
@@ -100,19 +90,41 @@ sub _getWmiSoftwareLicensingProducts {
             FULLNAME  => $object->{'Description'},
             NAME      => $object->{'Name'}
         };
-
-        if ($seenKey) {
-            # Update FULLNAME if seen ProductName in registry
-            $license->{FULLNAME} = encodeFromRegistry($seenKey->{'/ProductName'})
-                if $seenKey->{'/ProductName'};
-            $license->{TRIAL} = 1
-                if ($seenKey->{'/ProductNameBrand'} && $seenKey->{'/ProductNameBrand'} =~ /trial/i);
+                   
+        my $indexFoundLicence;
+        if ($object->{'ID'} && $seenProducts->{lc($object->{'ID'})}) {
+            my $seenKey = $seenProducts->{lc($object->{'ID'})};
+            if ($seenKey->{'/ProductCode'}) {
+                my $ProductCodeUuid =  lc( $seenKey->{'/ProductCode'} =~ /([-\w]+)/ && $1 );
+                # Check Office registry is true
+                if ($seenProducts->{$ProductCodeUuid} && $seenProducts->{$ProductCodeUuid}->{'/DigitalProductID'}) {
+                    # Find Office registry licence
+                    my $passLicence = 0;
+                    foreach my $seenLicence (@{$licences}) {
+                        if($seenLicence->{PRODUCTID} eq $seenProducts->{$ProductCodeUuid}->{'/ProductID'}) {
+                            $passLicence = 1 if $seenLicence->{KEY} =~ m/$partialProductKey$/;
+                            # $passLicence = 0, so the registry calculation of the key is wrong, WMI information are more revelant, so we delete the wrong Licence Office Registry
+                            ($indexFoundLicence) = grep { @{$licences}[$_] eq $seenLicence } 0..$#$licences if !$passLicence;
+                            last;
+                        }
+                    }
+                    # $passLicence = 1, Registry Licence is good, it supply more information, so we skip WMI Office Licence
```

Comment:
In fact, I don't like the way you decide to update or not a license. Here I would expect you found the license with a grep or eventually using `first` API from `FusionInventory::Agent::Tools`. Then in place of replacing the license at L.123, I would prefer you uses a private _mergeLicense() method passing the found and the new one as hash ref. In that method, you can handle what to merge as expected in the ref found in the current licenses list. Of course, the merge could be done from L.123 if you prefer. But without a loop here, you still can merge and do a next.